File: service/app/src/main/java/io/miragon/blueprint/adapter/outbound/db/LeasingApplicationJpaRepository.java
package io.miragon.blueprint.adapter.outbound.db;

import io.miragon.blueprint.domain.leasing.LeasingStatus;
import java.util.Optional;
import java.util.UUID;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

public interface LeasingApplicationJpaRepository extends JpaRepository<LeasingApplicationEntity, UUID> {

    Optional<LeasingApplicationEntity> findByApplicationId(UUID id);

    Page<LeasingApplicationEntity> findAllByStatus(LeasingStatus status, Pageable pageable);
}

File: service/app/src/main/kotlin/io/miragon/blueprint/adapter/outbound/db/LeasingApplicationJpaRepository.kt
package io.miragon.blueprint.adapter.outbound.db

import io.miragon.blueprint.domain.leasing.LeasingStatus
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

interface LeasingApplicationJpaRepository : JpaRepository<LeasingApplicationEntity, UUID> {
    fun findByApplicationId(id: UUID): LeasingApplicationEntity?

    fun findAllByStatus(status: LeasingStatus, pageable: Pageable): Page<LeasingApplicationEntity>
}
